protos/transaction.proto

```diff
@@ -71,10 +71,15 @@ message Transaction {
     //
     // Fragment IDs are not yet assigned.
     repeated DataFragment fragments = 1;
+    // The new indices
+    //
+    // Currently only provided when overwriting with an
+    // old version of the dataset.
+    repeated IndexMetadata indices = 2;
```

Comment:
This is a backwards incompatible protobuf change which should generally be frowned upon but I think(?) we haven't released the feature yet?

If we have then we should probably reverse this so `indices=4`

Reply:
we last released 3 days ago but i think transactions was merged 2 days ago

Reply:
one question though is whether indices are excluded from the transactions by design or not? I noticed that none of the transactions contain index metadata, even though for a number of operations index can be optionally inherited.

cc @wjones127

Reply:
Also, continuing from the slack conversation. Seems like the index metadata is handled inside the transaction commit process (see commit.rs and transaction.rs)

Reply:
So the question here is what kind of commit operation should this be? Overwrite discards the indices. Is this a `Rewrite` operation? or a new Restore commit operation? cc @wjones127 @chebbyChefNEQ

Reply:
I think we could just add a `Restore {to_version:...}` transaction, which should conflict with all other transactions. Then, in `build_manifest` we can checkout the `to_version` and write a new version the same as the `to_version` manifest.

Reply:
This is roughly what the restore version will look like.  I still need to tinker with the indices (and probably add a test): https://github.com/lancedb/lance/compare/main...westonpace:lance:experiment/rebase-restore-on-tx-2?expand=1

Reply:
My intention that indices behavior was determined by transaction type. Append -> keep them. Overwrite -> drop them. Restore should be it's own transaction type I think.